packages/eservice-template-process/src/model/domain/toEvent.ts

```diff
@@ -0,0 +1,472 @@
+  EServiceDocumentId,
+  AttributeId,
+} from "pagopa-interop-models";
+
+export const toCreateEventEServiceTemplateAdded = (
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId: eserviceTemplate.id,
+  version: 0,
+  event: {
+    type: "EServiceTemplateAdded",
+    event_version: 2,
+    data: { eserviceTemplate: toEServiceTemplateV2(eserviceTemplate) },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisAdded = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisAdded",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisDeleted = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisDeleted",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisUpdated = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisUpdated",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateDraftUpdated = (
+  streamId: string,
+  version: number,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateDraftUpdated",
+    event_version: 2,
+    data: {
+      eserviceTemplateId: eserviceTemplate.id,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateDraftVersionUpdated = (
+  streamId: string,
+  version: number,
+  eserviceTemplateVersionId: EServiceTemplateVersionId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateDraftVersionUpdated",
+    event_version: 2,
+    data: {
+      eserviceTemplateVersionId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
```

Comment:
same of previous comment

Reply:
The eservice template version id should stay in order to have the information of which template version changed